xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -751,6 +751,15 @@ void CActiveAESink::EnumerateOutputDevices(AEDeviceList &devices, bool passthrou
       if (passthrough && devInfo.m_deviceType == AE_DEVTYPE_PCM)
         continue;
 
+      // filters devices that should not be shown in the list
+      // of AUDIO DEVICES or AUDIO PASSTHROUGH DEVICES
+      // according to the capabilities informed by each sink
+      if (devInfo.m_onlyPassthrough && !passthrough)
+        continue;
+
+      if (devInfo.m_onlyPCM && passthrough)
```

Comment:
We don't need m_onlyPCM - this was what AE_DEVTYPE_PCM meant to be, right?

Reply:
I'm not sure: think are two different concepts. Device type can be `AE_DEVTYPE_HDMI`  and at same time used for PCM (PCM/not passthrough over HDMI)

The new concept if is the sink "type"  can not be used for PT / no PT.

DirectSound are reported at same time as PCM device e.g. Realtek motherboard audio and HDMI device on Nvidia graphics.

Reply:
Yes.

Before:
AE_DEVTYPE_HDMI -> PCM + PT
AE_DEVTYPE_SPDIF -> PCM + PT (2 channels)
AE_DEVTYPE_PCM -> PCM only.

So the functionality that is needed here, could be:
pcm only device: AE_DEVTYPE_PCM
PT only device: (AE_DEVTYPE_HDMI or AE_DEVTYPE_SPDIF)  && m_onlyPT.